src/cancelContentPlay.js

```diff
@@ -14,7 +14,7 @@ const cancelContentPlay = function(player) {
   }
 
   // Avoid content flash on non-iPad iOS
-  if (videojs.browser.IS_IOS && videojs.browser.IS_IPHONE) {
+  if ((videojs.browser.IS_IOS && videojs.browser.IS_IPHONE) && (!player.el_.hasAttribute('playsinline'))) {
```

Comment:
I don't think that the inner parens are necessary, at least on the `hasAttribute` part.